be/src/formats/parquet/stored_column_reader.cpp

```diff
@@ -111,6 +111,19 @@ class OptionalStoredColumnReader : public StoredColumnReader {
         *num_levels = _levels_parsed;
     }
 
+    void append_default_levels(size_t row_nums) {
+        if (_need_parse_levels) {
+            size_t new_capacity = _levels_parsed + row_nums;
+            if (new_capacity > _levels_capacity) {
+                _def_levels.resize(new_capacity);
```

Comment:
do you have to resize only when capacity grows, or do you have to resize to `new_capacity` anyway?